src/ray/core_worker/task_manager.cc

```diff
@@ -1420,14 +1420,10 @@ void TaskManager::MarkTaskWaitingForExecution(const TaskID &task_id,
 }
 
 void TaskManager::MarkTaskRetryOnResubmit(TaskEntry &task_entry) {
-  // Record the old attempt status as FINISHED.
```

Comment:
If we don't mark it as finished/failed, what status does the old task attempt have?

Reply:
The old task attempt should have already been marked as FINISHED. Resubmit is for lineage reconstruction, it doesn't always happen and can happen much later after the previous attempt finishes.